client/layout/index.jsx

```diff
@@ -261,7 +262,9 @@ class Layout extends Component {
 			( this.props.isGlobalSidebarVisible && this.props.colorScheme !== 'global' ) ||
 			( prevProps.isGlobalSidebarVisible &&
 				! this.props.isGlobalSidebarVisible &&
-				this.props.colorScheme !== 'global' )
+				this.props.colorScheme !== 'global' ) ||
+			( prevProps.sectionName !== 'checkout' && this.props.sectionName === 'checkout' ) ||
+			( prevProps.sectionName === 'checkout' && this.props.sectionName !== 'checkout' )
```

Comment:
This is a bit complicated to read, I think naming it before the `if` would be better:

```
const isCurrentOrPreviousSectionCheckout = ( prevProps.sectionName !== 'checkout' && this.props.sectionName === 'checkout' ) ||
			( prevProps.sectionName === 'checkout' && this.props.sectionName !== 'checkout' )
```

Reply:
I'll change it!